src/walle-lib/Obstacle.hpp

```diff
@@ -0,0 +1,26 @@
+//
+// Created by tonys on 28/11/2024.
+//
+
+#ifndef OBSTACLE_HPP
```

Comment:
You are mixing **#pragma once** and #ifndef, 
Please unify. 
#pragma once is non standard but supported by msvc, clang and gcc. Its advantage is that is is a single line, reduce maintenance and there is no risk of misspelling. 

While 
#ifndef XXXX
#define XXXX
...
#endif /// XXXX
Is longer to type, it also need cares to avoid having multiple .h files with the same XXXX and maintenaing the #if and #end is harder. 
In SOFA we switch to #pragma.